torchmetrics/functional/text/sacre_bleu.py

```diff
@@ -0,0 +1,354 @@
+
+        return cls._tokenize_regex(line)
+
+    @classmethod
+    def _tokenize_zh(cls, line: str) -> str:
+        """The tokenization of Chinese text in this script contains two
+        steps: separate each Chinese characters (by utf-8 encoding); tokenize
+        the non Chinese part (following the `13a` i.e. mteval tokenizer).
+        Author: Shujian Huang huangsj@nju.edu.cn
+
+        Args:
+            line: input sentence
+
+        Return:
+            tokenized sentence
+        """
+
+        line = line.strip()
+        line_in_chars = ""
+
+        for char in line:
+            if cls._is_chinese_char(char):
+                line_in_chars += " "
+                line_in_chars += char
+                line_in_chars += " "
+            else:
+                line_in_chars += char
+
+        return cls._tokenize_regex(line_in_chars)
+
+    @classmethod
+    def _tokenize_international(cls, line: str) -> str:
+        """Tokenizes a string following the official BLEU implementation.
+
+        See github.com/moses-smt/mosesdecoder/blob/master/scripts/generic/mteval-v14.pl#L954-L983
+
+        In our case, the input string is expected to be just one line.
+        We just tokenize on punctuation and symbols,
+        except when a punctuation is preceded and followed by a digit
+        (e.g. a comma/dot as a thousand/decimal separator).
+        We do not recover escaped forms of punctuations such as &apos; or &gt;
+        as these should never appear in MT system outputs (see issue #138)
+
+        Note that a number (e.g., a year) followed by a dot at the end of
+        sentence is NOT tokenized, i.e. the dot stays with the number because
+        `s/(\\p{P})(\\P{N})/ $1 $2/g` does not match this case (unless we add a
+        space after each sentence). However, this error is already in the
+        original mteval-v14.pl and we want to be consistent with it.
+        The error is not present in the non-international version,
+        which uses `$norm_text = " $norm_text "`.
+
+        Args:
+            line: the input string to tokenize.
+
+        Return:
+            The tokenized string.
+        """
+        for (_re, repl) in cls._INT_REGEX:
+            line = _re.sub(repl, line)
+
+        return " ".join(line.split())
+
+    @classmethod
+    def _tokenize_char(cls, line: str) -> str:
+        """Tokenizes all the characters in the input line.
+
+        Args:
+            line: a segment to tokenize
+        Return:
+            the tokenized line
+        """
+        return " ".join(char for char in line)
+
+    @staticmethod
+    def _lower(line: str, lowercase: bool) -> str:
+        if lowercase:
+            return line.lower()
+        return line
+
+
+def sacre_bleu_score(
+    reference_corpus: Sequence[Sequence[str]],
+    translate_corpus: Sequence[str],
+    n_gram: int = 4,
+    smooth: bool = False,
+    tokenize: Literal["none", "13a", "zh", "intl", "char"] = "13a",
```

Comment:
Thinking about using Enum for these cases, similar as we have StrEnum
https://github.com/PyTorchLightning/pytorch-lightning/blob/ab207921b9149a2c20ff9f797b6381f61cb09b1e/pytorch_lightning/utilities/enums.py#L19